### PR TITLE
feat: add core game elements

### PR DESCRIPTION
**Pull request by tranzystorekk**
Adds a reasonable base for a playable minesweeper:

- state handling
- graphics + fonts
- ui: game timer, `flags placed / total mines` indicator

Supersedes #6

### CODE REVIEW

File: src/drawing.rs [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+const COVER_COLOR: Color = Color::GRAY;
+const FLAG_COLOR: Color = Color::RED;
+const UNSURE_COLOR: Color = Color::BLUE;
+
+pub fn draw(gfx: &mut Graphics, state: &mut State) {
+    let mut draw = gfx.create_draw();
+
+    draw.clear(Color::BLACK);
+
+    draw_ui(&mut draw, state);
```
Comment (HalfInner) [inline src/drawing.rs:28]: Shouldn't the UI be drawn last? (after board)

File: src/drawing.rs [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+        }
+        Stage::Victory => {
+            if let Cover::Up(_) = cover {
+                fill_color = WIN_COLOR;
+            }
+        }
+        _ => (),
+    }
+
+    // adjust color for mouse hover
```
Comment (HalfInner) [inline src/drawing.rs]: Short function's always better than comment

File: src/drawing.rs [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+
+    draw.rect(pos, DIMS).color(fill_color);
+    draw.rect(pos, DIMS).color(OUTLINE_COLOR).stroke(STROKE);
+
+    // draw hint number
+    if let (Cover::Down, Object::Hint(n)) = (cover, object) {
+        draw.text(state.font(), &n.to_string())
+            .color(Color::BLACK)
+            .size(26.0)
+            .position(screen_x + TILE_SIZE * 0.5, screen_y + TILE_SIZE * 0.5)
```
Comment (HalfInner) [inline src/drawing.rs]: Declare HALF_TILE_SIZE

File: src/state/victory.rs [hunk 4]
```diff
@@ -0,0 +1,13 @@
+use notan::prelude::*;
+
+use crate::state::{Stage, State};
+
+pub fn update(app: &mut App, state: &mut State) {
+    state.hover = None;
+
+    if app.keyboard.was_pressed(KeyCode::Space) {
+        state.stage = Stage::Playing;
+        state.elapsed_milisec = 0;
```
Comment (HalfInner) [inline src/state/victory.rs]: I'd rename elapsed_milisec to something more applicable to total player/game time

File: src/drawing.rs (hunk 1 above)
Comment (tranzystorekk) [reply src/drawing.rs:28]: For now the UI does not overlap with the minefield board, so it makes no difference

File: src/state/victory.rs (hunk 4 above)
Comment (tranzystorekk) [reply src/state/victory.rs]: Maybe `run_timer_milisec`?

File: src/drawing.rs (hunk 2 above)
Comment (tranzystorekk) [reply src/drawing.rs]: Not sure if it's worth extracting the minuscule amount of code, but I agree this comment looks unnecessary

**Pull request merged**